components/filecoin/rs/src/signature.rs

```diff
@@ -0,0 +1,190 @@
+use crate::message::MessageAPI;
+use blake2b_simd::Params;
+use bls_signatures::Serialize;
+use core::{array::TryFromSliceError, num::ParseIntError};
+use fvm_ipld_encoding::Cbor;
+use fvm_shared::address::Protocol;
+use fvm_shared::message::Message as UnsignedMessage;
+use libsecp256k1::util::{SECRET_KEY_SIZE, SIGNATURE_SIZE};
+use libsecp256k1::{sign, Message};
+use std::convert::TryInto;
+use thiserror::Error;
+
+pub const SIGNATURE_RECOVERY_SIZE: usize = SIGNATURE_SIZE + 1;
+pub const BLS_SIGNATURE_SIZE: usize = 96;
+
+pub struct PrivateKey(pub [u8; SECRET_KEY_SIZE]);
+pub struct SignatureBLS(pub [u8; BLS_SIGNATURE_SIZE]);
+pub struct SignatureSECP256K1(pub [u8; SIGNATURE_RECOVERY_SIZE]);
+
+struct SecpSignature {
+  signature: libsecp256k1::Signature,
+  recid: libsecp256k1::RecoveryId
+}
+
+pub enum Signature {
+    SignatureSECP256K1(SignatureSECP256K1),
+    SignatureBLS(SignatureBLS),
+}
+
+impl Signature {
+    pub fn as_bytes(&self) -> &[u8] {
+        match self {
+            Signature::SignatureSECP256K1(sig_secp256k1) => sig_secp256k1.as_bytes(),
+            Signature::SignatureBLS(sig_bls) => sig_bls.as_bytes(),
+        }
+    }
+}
+
+impl SignatureBLS {
+    pub fn as_bytes(&self) -> &[u8] {
+        &self.0
+    }
+}
+
+impl SignatureSECP256K1 {
+    pub fn as_bytes(&self) -> &[u8] {
+        &self.0
+    }
+}
+
+/// Filecoin Signer Error
+/// https://github.com/Zondax/filecoin-signing-tools/blob/2e7b005b4733761de11d4a252cb481ca5aedb029/signer/src/error.rs#L7
+#[derive(Error, Debug)]
+pub enum SignerError {
+    /// Secp256k1 error
+    #[error("secp256k1 error")]
+    Secp256k1(#[from] libsecp256k1::Error),
+    /// InvalidBigInt error
+    #[error("InvalidBigInt error")]
+    InvalidBigInt(#[from] num_bigint_chainsafe::ParseBigIntError),
+    /// Generic error message
+    #[error("Error: `{0}`")]
+    GenericString(String),
+    /// Not able to parse integer
+    #[error("Cannot parse integer")]
+    ParseIntError(#[from] ParseIntError),
+    /// BLS error
+    #[error("BLS error | {0}")]
+    BLS(#[from] bls_signatures::Error),
+    /// Invalid BIP44Path
+    #[error("Invalid BIP44 path : `{0}`")]
+    InvalidBIP44Path(#[from] zx_bip44::errors::BIP44PathError),
```

Comment:
We're not throwing this error ourself from what I can tell so I think we can drop this and the dependency

Reply:
done